packages/react-data-grid/src/masks/InteractionMasks.js

```diff
@@ -296,23 +315,60 @@ class InteractionMasks extends React.Component {
 
   changeCellFromEvent(e) {
     e.preventDefault();
+    const isTab = e.keyCode === keyCodes.Tab;
+    const isShift = e.shiftKey;
+
+    if (isTab) {
+      this.changeCellFromTabAction(e);
+    } else if (isShift) {
+      this.changeSelectedRangeFromArrowKeyAction(e);
+    } else {
+      this.changeCellFromArrowKeyAction(e);
+    }
+  }
+
+  changeCellFromArrowKeyAction(e) {
+    const currentPosition = this.state.selectedPosition;
+    const { cellNavigationMode } = this.props;
+    const next = this.getNextPositionAndFireOnHitBoundaryIfNeccesary(e, currentPosition, cellNavigationMode);
```

Comment:
The name of this function suggests it has too many responsibilities
Better change this to 
```javascript
const next = this.getNextPosition(e, currentPosition, cellNavigationMode);
this.checkCellIsAtGridBoundary();
```